File: app/src/main/java/com/implozia/cogitare/ui/MainViewModel.java
package com.implozia.cogitare.ui;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;


import com.implozia.cogitare.App;
import com.implozia.cogitare.model.Note;

import java.util.List;

public class MainViewModel extends ViewModel {
    private LiveData<List<Note>> noteLiveData = App.getInstance().getNoteDao().getAllLiveData();

    public LiveData<List<Note>> getNoteLiveData() {
        return noteLiveData;
    }
}

File: app/src/main/java/com/implozia/cogitare/presentation/ui/main/MainViewModel.kt
package com.implozia.cogitare.presentation.ui.main

import androidx.lifecycle.ViewModel
import com.implozia.cogitare.data.repository.NoteRepository
import com.implozia.cogitare.model.Note

class MainViewModel(private val noteRepository: NoteRepository) : ViewModel() {
    suspend fun noteEntries() = noteRepository.getAllNote()
    suspend fun updateEntries(note: Note) = noteRepository.update(note)
    suspend fun insertEntries(note: Note) = noteRepository.insert(note)
}